src/AsyncEventDispatcher.php

```diff
@@ -120,14 +130,34 @@ private function dispatchNonStoppableEvent(
             foreach ($listeners as $listener) {
                 $futures[] = async(function () use ($event, $listener) {
                     $listener($event);
-                });
+                })->catch(Closure::fromCallable([$this, 'errorHandler']));
             }
 
             // Wait for all listeners to complete
-            awaitAll($futures, $cancellation);
+            if ($this->options & self::THROW_ON_ERROR) {
+                // Let the errors bubble up
+                await($futures, $cancellation);
+            } else {
+                // Carry on despite errors
+                awaitAll($futures, $cancellation);
```

Comment:
I would personally remove this option and just always `awaitAll()` 🤷

Reply:
Went back to always using `awaitAll()`